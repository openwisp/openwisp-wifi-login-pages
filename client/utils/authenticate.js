import {sessionStorage} from "./storage";

const authenticate = (cookies, orgSlug) => {
  const token = cookies.get(`${orgSlug}_auth_token`);
  const sessionKey = sessionStorage.getItem(`${orgSlug}_auth_token`);
  if (sessionKey) {
    cookies.remove(`${orgSlug}_auth_token`, {path: "/"});
    cookies.remove(`${orgSlug}_username`, {path: "/"});
    return true;
  }
  if (token) return true;
  return false;
};
export default authenticate;
