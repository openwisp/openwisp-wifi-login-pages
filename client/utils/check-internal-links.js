const isInternalLink = (link) => {
  if (link.length && link[0] === "/") return true;
  return false;
};
export default isInternalLink;
