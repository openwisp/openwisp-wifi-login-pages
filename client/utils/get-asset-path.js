const getAssetPath = (slug, relativePath) => `/assets/${slug}/${relativePath}`;
export default getAssetPath;
