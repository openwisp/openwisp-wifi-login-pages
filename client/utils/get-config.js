const loadConfig = (slug) => import(`../configs/${slug}.json`);
const getConfig = async (slug) => {
  let config;
  try {
    config = await loadConfig(slug);
  } catch (err) {
    console.error(`Configuration for organization ${slug} does not exists.`);
    config = undefined;
  }
  return config;
};

export default getConfig;
