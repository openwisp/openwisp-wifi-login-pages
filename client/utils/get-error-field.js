import React from "react";

const getErrorField = (errors, key, nonField = false, className = "") =>
  errors[key] ? (
    <div className={!nonField ? `error ${className}` : "error non-field"}>
      <span className="icon">!</span>
      <span className="text">{errors[key]}</span>
    </div>
  ) : null;

export default getErrorField;
