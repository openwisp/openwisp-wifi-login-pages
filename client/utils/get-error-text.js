const getErrorText = (error, fallbackVal = "") => {
  const {data} = error.response;
  // eslint-disable-next-line no-nested-ternary
  return data.detail
    ? data.detail
    : data.non_field_errors
    ? data.non_field_errors[0]
    : fallbackVal;
};
export default getErrorText;
