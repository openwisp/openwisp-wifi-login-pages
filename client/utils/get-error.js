import React from "react";

const getError = (errors, key = null) => {
  const errorKey = key === null ? "nonField" : key;
  return errors[errorKey] ? (
    <div className={key !== null ? `error ${key}` : "error non-field"}>
      <span className="icon" />
      <span className="text">{errors[errorKey]}</span>
    </div>
  ) : null;
};

export default getError;
