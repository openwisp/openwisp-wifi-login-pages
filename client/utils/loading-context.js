import React from "react";

export const loadingContextValue = {setLoading: () => {}, getLoading: () => {}};
const LoadingContext = React.createContext(loadingContextValue);

export default LoadingContext;
