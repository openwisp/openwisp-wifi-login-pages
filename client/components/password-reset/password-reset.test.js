/* eslint-disable prefer-promise-reject-errors */
import axios from "axios";
import {shallow} from "enzyme";
import React from "react";
import {toast} from "react-toastify";
import PropTypes from "prop-types";
import getConfig from "../../utils/get-config";
import {loadingContextValue} from "../../utils/loading-context";
import loadTranslation from "../../utils/load-translation";
import PasswordReset from "./password-reset";
import translation from "../../test-translation.json";

jest.mock("axios");
jest.mock("../../utils/get-config");
jest.mock("../../utils/load-translation");

const defaultConfig = getConfig("default", true);
const createTestProps = (props) => ({
  orgSlug: "default",
  orgName: "default name",
  setTitle: jest.fn(),
  passwordReset: defaultConfig.components.password_reset_form,
  language: "en",
  ...props,
});

const getTranslationString = (msgid) => {
  try {
    return translation.translations[""][msgid].msgstr[0];
  } catch (err) {
    // eslint-disable-next-line no-console
    console.error(err, msgid);
    return msgid;
  }
};

describe("<PasswordReset /> rendering with placeholder translation tags", () => {
  const props = createTestProps();
  it("should render translation placeholder correctly", () => {
    const wrapper = shallow(<PasswordReset {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<PasswordReset /> rendering", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    loadTranslation("en", "default");
    wrapper = shallow(<PasswordReset {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 2 inputs", () => {
    expect(wrapper.find("input")).toHaveLength(2);
    expect(wrapper.find("input[type='email']")).toHaveLength(1);
    expect(wrapper.find("input[type='submit']")).toHaveLength(1);
  });

  it("should render email field correctly", () => {
    const emailInput = wrapper.find("input[type='email']");
    expect(wrapper.find(".row.email label").text()).toBe(
      getTranslationString("EMAIL"),
    );
    expect(emailInput.prop("placeholder")).toBe(
      getTranslationString("EMAIL_PHOLD"),
    );
    expect(emailInput.prop("title")).toBe(
      getTranslationString("EMAIL_PTRN_DESC"),
    );
    expect(emailInput.prop("type")).toBe("email");
  });
});

describe("<PasswordReset /> interactions", () => {
  let props;
  let wrapper;
  let originalError;
  let lastConsoleOutuput;

  beforeEach(() => {
    originalError = console.error;
    lastConsoleOutuput = null;
    console.error = (data) => {
      lastConsoleOutuput = data;
    };
    props = createTestProps();
    PasswordReset.contextTypes = {
      setLoading: PropTypes.func,
      getLoading: PropTypes.func,
    };
    wrapper = shallow(<PasswordReset {...props} />, {
      context: loadingContextValue,
    });
  });

  afterEach(() => {
    console.error = originalError;
  });

  it("should change state values when handleChange function is invoked", () => {
    wrapper
      .find("input[type='email']")
      .simulate("change", {target: {value: "test@test.com", name: "email"}});
    expect(wrapper.state("email")).toEqual("test@test.com");
  });

  it("should execute handleSubmit correctly when form is submitted", () => {
    axios
      .mockImplementationOnce(() =>
        Promise.reject({response: {data: {detail: "errors"}}}),
      )
      .mockImplementationOnce(() =>
        Promise.reject({
          response: {data: {non_field_errors: ["non field errors"]}},
        }),
      )
      .mockImplementationOnce(() => Promise.resolve({data: {detail: true}}));
    const spyToastError = jest.spyOn(toast, "error");
    const spyToastSuccess = jest.spyOn(toast, "success");
    return wrapper
      .instance()
      .handleSubmit({preventDefault: () => {}})
      .then(() => {
        expect(wrapper.instance().state.errors.email).toEqual("errors");
        expect(wrapper.find("div.error")).toHaveLength(1);
        expect(wrapper.find("input.error")).toHaveLength(1);
        expect(lastConsoleOutuput).not.toBe(null);
        expect(spyToastError.mock.calls.length).toBe(1);
        expect(spyToastSuccess.mock.calls.length).toBe(0);
        lastConsoleOutuput = null;
      })
      .then(() =>
        wrapper
          .instance()
          .handleSubmit({preventDefault: () => {}})
          .then(() => {
            expect(wrapper.instance().state.errors.email).toEqual(
              "non field errors",
            );
            expect(lastConsoleOutuput).not.toBe(null);
            expect(spyToastError.mock.calls.length).toBe(2);
            expect(spyToastSuccess.mock.calls.length).toBe(0);
            lastConsoleOutuput = null;
          }),
      )
      .then(() =>
        wrapper
          .instance()
          .handleSubmit({preventDefault: () => {}})
          .then(() => {
            expect(wrapper.instance().state.errors).toEqual({});
            expect(wrapper.instance().state.success).toBe(true);
            expect(wrapper.find(".error")).toHaveLength(0);
            expect(wrapper.find(".success")).toHaveLength(1);
            expect(lastConsoleOutuput).toBe(null);
            expect(spyToastError.mock.calls.length).toBe(2);
            expect(spyToastSuccess.mock.calls.length).toBe(1);
            lastConsoleOutuput = null;
          }),
      );
  });
  it("should set title", () => {
    const setTitleMock = wrapper.instance().props.setTitle.mock;
    expect(setTitleMock.calls.pop()).toEqual(["Reset Password", props.orgName]);
  });
});
