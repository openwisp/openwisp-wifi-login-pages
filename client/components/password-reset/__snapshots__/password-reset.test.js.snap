// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordReset /> rendering should render correctly 1`] = `
<div
  className="container content"
  id="reset-password"
>
  <div
    className="inner"
  >
    <form
      className="main-column"
      onSubmit={[Function]}
    >
      <div
        className="inner"
      >
        <p
          className="label"
        >
          Have you forgot your password? Enter your email address below and we'll send you the instructions to reset it.
        </p>
        <div
          className="fieldset"
        >
          <div
            className="row email"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              autoComplete="email"
              className="input "
              id="email"
              name="email"
              onChange={[Function]}
              pattern=".+@.+\\\\..+"
              placeholder="email address"
              required={true}
              title="enter valid email address"
              type="email"
              value=""
            />
          </div>
          <div
            className="row submit"
          >
            <input
              className="button full"
              type="submit"
              value="Reset My Password"
            />
          </div>
        </div>
        <div
          className="row contact-us"
        >
          Please contact us if you have any trouble resetting your password.
        </div>
        <div
          className="row links"
        >
          <Link
            className="link"
            to="/default/login"
          >
            Go back to sign in
          </Link>
        </div>
      </div>
    </form>
    <Connect(Contact) />
  </div>
</div>
`;

exports[`<PasswordReset /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<div
  className="container content"
  id="reset-password"
>
  <div
    className="inner"
  >
    <form
      className="main-column"
      onSubmit={[Function]}
    >
      <div
        className="inner"
      >
        <p
          className="label"
        >
          PWD_RESET_ADD_TXT
        </p>
        <div
          className="fieldset"
        >
          <div
            className="row email"
          >
            <label
              htmlFor="email"
            >
              EMAIL
            </label>
            <input
              autoComplete="email"
              className="input "
              id="email"
              name="email"
              onChange={[Function]}
              pattern=".+@.+\\\\..+"
              placeholder="EMAIL_PHOLD"
              required={true}
              title="EMAIL_PTRN_DESC"
              type="email"
              value=""
            />
          </div>
          <div
            className="row submit"
          >
            <input
              className="button full"
              type="submit"
              value="PWD_RESET_BTN"
            />
          </div>
        </div>
        <div
          className="row contact-us"
        >
          PWD_RESET_CNTC_TXT
        </div>
        <div
          className="row links"
        >
          <Link
            className="link"
            to="/default/login"
          >
            LOGIN_PG_LNK
          </Link>
        </div>
      </div>
    </form>
    <Connect(Contact) />
  </div>
</div>
`;
